Features/Context/SubContext/CommonActions.php

```diff
@@ -0,0 +1,363 @@
+
+        if ( is_string( $result ) )
+        {
+            return $this->clickContentTreeLink( $dir, ++$index, $result );
+        }
+    }
+
+    /**
+     * Makes and element visible so a file can be attached to it
+     *
+     * @param string $path      file path relative to mink definitions
+     * @param string $selector  selector of the element do to make visible
+     */
+    protected function attachFilePrepare( $path, $selector )
+    {
+        $jsArgs = JsHelper::generateFuncArgs( $selector );
+        $jsCode = "return BDD.changeCssDisplay( $jsArgs );";
+        $this->execJavascript( $jsCode );
+
+        $this->attachFile( $path );
+    }
+
+    /**
+     * Attaches a file to a input field en HTML
+     *
+     * @param string $path      file path relative to mink definitions
+     */
+    protected function attachFile( $path )
+    {
+        if ( $this->getMinkParameter( 'files_path' ) )
+        {
+            $fullPath = rtrim( realpath( $this->getMinkParameter( 'files_path' ) ), DIRECTORY_SEPARATOR ).DIRECTORY_SEPARATOR.$path;
+
+            if ( is_file( $fullPath ) )
+            {
+                $fileInput = 'input[type="file"]';
+                $field = $this->getSession()->getPage()->find( 'css', $fileInput );
+
+                if ( null === $field )
+                {
+                    throw new Exception( "File input is not found" );
+                }
+            $field->attachFile( $fullPath );
+            }
+        }
+        else throw new Exception( "File is not found at the given location" );
+    }
+
+    /**
+     * Drops a file to an HTML element with specific text
+     *
+     * @param string $path      file path relative to mink definitions
+     * @param string $selector  selector of the element where to drop the file
+     */
+    protected function dragAndDropFile( $path, $selector )
+    {
+        if ( $this->getMinkParameter( 'files_path' ) )
+        {
+            $fullPath = rtrim( realpath( $this->getMinkParameter( 'files_path' ) ), DIRECTORY_SEPARATOR ).DIRECTORY_SEPARATOR.$path;
+
+            if ( is_file( $fullPath ) )
+            {
+                $fileName = $path;
+                $finfo = finfo_open( FILEINFO_MIME_TYPE );
+                $fileType = finfo_file( $finfo, $fullPath );
+                $contents = file_get_contents( $fullPath );
+                $contentBase64 = base64_encode( $contents );
+
+                $jsArgs = JsHelper::generateFuncArgs( $fileName, $fileType, $contentBase64, $selector );
+                $jsCode = "return BDD.simulateDropEventWithFile( $jsArgs );";
+                $jsFile = $this->execJavascript( $jsCode );
+            }
+        }
+        else throw new Exception( "File is not found at the given location" );
+    }
+
+    /**
+     * Finds an HTML element by class and the text value and return the value of the sibling
+     *
+     * @param string $name
+     */
+    protected function checksImagePresent( $text, $selector )
```

Comment:
The text is needed to find the image, normally associated with a text label, example:
![image](https://cloud.githubusercontent.com/assets/8983122/6462859/c110484e-c1a4-11e4-83d8-f31732b585c9.png)

$text in red